image/layer.go

```diff
@@ -0,0 +1,58 @@
+// Copyright 2016 The Linux Foundation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package image
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+
+	"github.com/opencontainers/image-tools/utils"
+)
+
+// CreateLayer cretes filesystem changset from child and parent
+func CreateLayer(child, parent, dest string) error {
+	arch, err := Diff(child, parent)
```

Comment:
I would somewhat like to use @vbatts' [go-mtree](https://github.com/vbatts/go-mtree) here. To be fair, I've been slacking on the required PRs to implement it, but would you be open to switching to go-mtree once it's up to snuff? The main benefit is that it has far more explicit design constraints about reproducibility and manifest verifiability than `pkg/archive`.